Feature/dossier service/complete

ev1gilatio opened the pull request.

Commit:
Initial commit for dossier-service

Commit:
Initial commit for dossier-service

Commit:
Закончил dossier-service

Commit:
1. Косметические изменения кода
2. Отрабатывает вся логика в соответствии со схемой из ТЗ
3. Доделан dossier-service

ev1gilatio commented on deal-service/src/main/java/ru/evig/dealservice/controller/DealController.java:
```diff
@@ -39,30 +46,158 @@ private List<LoanOfferDto> getLoanOfferList(@Valid @RequestBody LoanStatementReq
+
+//    @Operation(summary = "Получение кредитных документов",
+//            description = "Служебный эндпоинт для получения кредитных документов dossier-service'ом, " +
+//                    "напрямую сюда обращаться не нужно")
```
Случайно закомментировал и забыл раскомментировать

Sermjazhko replied:
Исправь

Sermjazhko commented on deal-service/src/main/java/ru/evig/dealservice/controller/DealController.java:
```diff
@@ -39,30 +46,158 @@ private List<LoanOfferDto> getLoanOfferList(@Valid @RequestBody LoanStatementReq
+            service.changeStatementStatus(statementId, ApplicationStatus.CC_DENIED);
+
+            throw new RuntimeException(e.contentUTF8());
+        }
```
Ты можешь не создавать новое исключение, а пробрасывать старое.

Sermjazhko commented on deal-service/src/main/java/ru/evig/dealservice/controller/DealController.java:
```diff
@@ -39,30 +46,158 @@ private List<LoanOfferDto> getLoanOfferList(@Valid @RequestBody LoanStatementReq
     @PostMapping("/calculate/{statementId}")
     private void createCredit(@Valid @RequestBody FinishRegistrationRequestDto frrDto,
                               @PathVariable String statementId) {
         ScoringDataDto sdDto = service.getScoringDataDto(frrDto, statementId);
```
Вынеси всю логику из контроллера в сервис

Sermjazhko commented on deal-service/src/main/java/ru/evig/dealservice/controller/DealController.java:
```diff
@@ -39,30 +46,158 @@ private List<LoanOfferDto> getLoanOfferList(@Valid @RequestBody LoanStatementReq
-        CreditDto dealClientCreditDto = dealClient.getCreditDto(sdDto);
-        service.completeBuildingClient(frrDto, statementId);
-        service.createCredit(dealClientCreditDto, statementId);
+
```
Вынеси всю логику из контроллера в сервис

Sermjazhko commented on deal-service/src/main/java/ru/evig/dealservice/controller/DealController.java:
```diff
@@ -39,30 +46,158 @@ private List<LoanOfferDto> getLoanOfferList(@Valid @RequestBody LoanStatementReq
+    })
+    @PostMapping("/document/{statementId}/sign")
+    private void requestForSigningDocuments(@PathVariable String statementId) {
+        service.checkDeniedStatus(statementId);
```
Посмотри, чтобы в каждом методе контроллера не было сервисной логики

Sermjazhko commented on deal-service/src/main/java/ru/evig/dealservice/controller/DealController.java:
```diff
@@ -9,25 +10,31 @@
-            @ApiResponse(responseCode = "400", description = "Какие-то данные для прескоринга неудовлетворяют условиям")
+            @ApiResponse(responseCode = "200", description = "Успешная валидация и обращение в calculator-service"),
+            @ApiResponse(responseCode = "400",description = "Какие-то данные не прошли валидацию " +
+                    "или прескоринг в calculator-service")
```
Вынеси всё описание в интерфейс, слишком нагружено

Sermjazhko commented on deal-service/src/main/java/ru/evig/dealservice/service/DealService.java:
```diff
@@ -10,18 +10,18 @@
+import java.util.*;
 
 @Service
 @Slf4j
```
Вынеси все создания объектов в соответствующий маппер (почему паспорт отдельно висит от клиент маппера?)

Sermjazhko commented on deal-service/src/main/java/ru/evig/dealservice/service/DealService.java:
```diff
@@ -57,8 +57,8 @@ public Statement createStatement(Client client) {
         Statement statement = Statement.builder()
                 .id(UUID.randomUUID())
```
Почему ты рандомишь руками UUID?

Sermjazhko commented on deal-service/src/main/java/ru/evig/dealservice/service/DealService.java:
```diff
@@ -198,4 +218,84 @@ private List<StatementStatusHistoryDto> getChangedStatementStatus(Statement stat
+                "Ставка: " + rate + "\n" +
+                "ПСК: " + psk + "\n" +
+                "Страховка: " + insurance + "\n" +
+                "Зарплатный клиент: " + salaryClient;
```
Почему логика создания документа в deal? И где формирование приложения к письму в виде документов?

Sermjazhko commented on statement-service/src/main/java/ru/evig/statementservice/controller/StatementController.java:
```diff
@@ -37,9 +38,10 @@ private List<LoanOfferDto> getLoanOfferList(@Valid @RequestBody LoanStatementReq
-            @ApiResponse(responseCode = "200", description = "Успешное отправление"),
+            @ApiResponse(responseCode = "200", description = "Все данные выбранного LoanOfferDto верны"),
             @ApiResponse(responseCode = "400", description = "Какие-то данные в LoanOfferDto неверны")
     })
```
Везде вынеси описания в интерфейс, чтобы не было так нагружено

Sermjazhko commented on dossier-service/src/main/java/ru/evig/dossierservice/service/EmailService.java:
```diff
@@ -0,0 +1,38 @@
+
+            log.info("Sent data to Email from sendEmail = " + message);
+        } catch (Exception e) {
+            e.printStackTrace();
```
Залогируй, вместо того чтобы стектрейс выкидывать

ev1gilatio committed:
1. Уменьшено количество вызовов методов сервиса из контроллеров
2. Добавлены мапперы для PassportDto и сущности Credit
3. Косметические изменения кода
4. Описание контроллеров вынесены в отделные интерфейсы
5. Добавлено формирование PDF файла с кредитными документами
6. Убран ID у PassportDto за ненадобностью

ev1gilatio committed:
Изменены структуры DealControllerImpl и DealService

ev1gilatio committed:
Убрал лишний импорт в DealControllerImpl
